### PR TITLE
Add a slack connector

### PR DESCRIPTION
**Pull request by hoxworth**

### CODE REVIEW

File: connectors/slack.rb [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+  def update_users(users)
+    puts "Updating users"
+    users.each do |user|
+      user_id = user["id"]
+      if !Linkbot.user_exists?(user_id)
+        Linkbot.add_user(user["name"],user_id)
+      else
+        Linkbot.update_user(user["name"],user_id)
+      end
```
Comment (robbkidd) [inline connectors/slack.rb]:
```ruby
if Linkbot.user_exists?(user_id)
  Linkbot.update_user(user["name"],user_id)
else
  Linkbot.add_user(user["name"],user_id)
end
```

**Comment by robbkidd** [comment]
Seems to hang together. I'd flip the `if..else` there in `update_users`, but otherwise :+1:

File: connectors/slack.rb [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    if Linkbot.user_exists?(nick)
+      message = Message.new(text, nick, Linkbot.username(nick), self, :message, options )
+      invoke_callbacks(message, options)
+    end
+  end
+
+  def send_messages(messages, options={})
+    messages.each do |message|
+      if message && message.strip.length > 0
+        @client.chat_postMessage({:channel => options[:room], :text => message})
```
Comment (llimllib) [inline connectors/slack.rb]: I think ```
@client.chat_postMessage({
  :channel => options[:room],
  :text => message,
  :username => "linkbot",
  :icon_emoji => ":punch:"
})
```

Would get you a "linkbot" name and a better icon

Comment (llimllib) [inline connectors/slack.rb]: https://github.com/aki017/slack-ruby-gem/blob/9960f6a13eedb790e6cfe75921d25eabce109c03/lib/slack/endpoint/chat.rb#L23

Comment (hoxworth) [inline connectors/slack.rb]: ah yes will change that tonight

**Comment by markolson** [comment]
imma test slack too

**Pull request merged**